gensim/models/keyedvectors.py

```diff
@@ -640,7 +641,7 @@ def evaluate_word_analogies_setBased(self, analogies, restrict_vocab=300000, cas
                 predicted = None
                 # find the most likely prediction using 3CosAvg set based vector offset) method
                 # Implementation of the set-based method for solving analogies
-                sims = self.most_similar(positive=terms[0:len(terms)-2:2], negative=terms[1:len(terms)-2:2], last=terms[len(terms)-2], topn=topk, restrict_vocab=restrict_vocab)
+                sims = self.most_similar(positive=terms[1:len(terms)-2:2], negative=terms[0:len(terms)-2:2], last=terms[len(terms)-2], topn=topk, restrict_vocab=restrict_vocab)
```

Comment:
Why start at `1`, why end at `len(terms) - 2`, go with step `2`?

This assumes `len(terms) >= 2`; why should that be?

Reply:
We have (a_i, b_i) word pairs in each row. For 11 pairs, we want to consider the average of the difference of the first ten pairs Σ(b_i - a_i)/10 and add it to the a11, and then using cosine similarity as in the 3CosAdd case try to find the b11 word. So the a_i words are the negative ones and the b_i are the positives. We are checking that we have complete pairs and there is not any word missing.

Reply:
len(terms) >= 2 needs to hold because otherwise it is a single word

Reply:
It starts at one because we cosinder the second word in each pair as the positive.

Reply:
Thanks, but all of this is too hard to read. Too much effort for maintainters (current and future) to decipher.

Please use comments, named variables and clear invariants:
* If an array *must* contain at least two items by design, use `assert` to make that clear (now it's not clear to me what the code does if that precondition is not fulfilled). If it's a user error instead (not a programmer error), raise a `ValueError`. 
* If some parts of an array hold special meaning, give them a special name / variable. For example, `assert len(terms) > 2`, `target_positive, target_negative = terms[ : -2]`, `positive_words = terms[1 : : 2]  # the second word in each pair is the positive one` etc (just examples).